rice-datanode/source/data_transfer_server.cc

```diff
@@ -36,12 +36,16 @@ void TransferServer::handle_connection(tcp::socket sock) {
 		}
 		// TODO: implement proto handlers based on type
 		switch (type) {
-			case (WRITE_BLOCK):
-				processWriteRequest(sock);
-				break;
-			case (READ_BLOCK):
-				processReadRequest(sock);
-				break;
+			case (WRITE_BLOCK): {
+                std::function <void(TransferServer&, tcp::socket&)> writeFn = &TransferServer::processWriteRequest;
+                synchronize(writeFn, sock);
+            }
+            break;
```

Comment:
I don't think it makes a difference. The braces are just to confine the *Fn variables.